### ISSUE TITLE
Question about hyperparameter tuning

### ISSUE THREAD

**Issue by DM-Berger** [high-dimensional/biobank-megamodeller#1]
From your (excellent, and technically impressive) paper, and from this repo code, I get the impression that a fixed learning rate of 1e-4 and some unspecified weight decay (seemingly a custom implementation with 1e-4) was used with Adam for ALL networks and predictions. Is there any reason the [properly functional AdamW](https://arxiv.org/abs/1711.05101) wasn't used instead? 

More concerningly, a fixed learning rate with no LR decay or warmup just, in my experience (which is limited admittedly to much smaller MRI and fMRI datasets like ABIDE, ABIDE-II, ADNI, ADHD-200), flat-out breaks many CNNs or other deep models that _would_ have worked well if given a tiny amount of tuning. E.g. a tiny grid that correctly ensures there is a functional combination of LRs and WDs (e.g. check LR in [1e-5, 1e-4, 1e-3] at bare minimum, and possibly also WD in [1e-5, 1e-4, 1e-3 ,1e-2] - it is enough to train for just a few epochs to check for non-divergence) can mean the difference between a predictive vs. non-predictive model, or the difference between near-SOTA vs barely-better-than-dummy-model performance . This would imply that your negative neuroimaging results could, perhaps, be interpreted as simply due to inadequate basic tuning / model misspecification. 

I find such a thing hard to believe, especially given the rest of the throughness and obvious familiarity with much of this whole MRI and deep learning ecosystem. Did you perchance do some undocumented hyperparameter sweep to establish that this LR and WD was not utterly broken for any models? Have I misread the paper in some way? Or are there training curves to demonstrate that e.g. early stopping wasn't just rapidly terminating hundreds of broken and misspecified models? 

It is also possible training was just far more stable for you, with 10000+ samples (where I was limited to less than 1000), so that hyperparameter tuning was not so important. I found these issues came up when attempting to predict psychological variables, but tuning was less important when I would try to predict age and sex, so, ultimately, I am inclined to agree with you and say that that this is not a tuning issue, but just that there isn't really enough information in the data here. But still, I'd be curious to know your thoughts on the relevance of hyperparameter tuning to your findings. 

Anyway, regardless, I really enjoyed the paper!

**Comment by jamesruffle** [comment]
Hi @DM-Berger,

Thank you for your kind words about the piece.

In the end, we trained with Adam, an _initial_ learning rate of 1e-4, and L2 regularisation. We did experiment with different LR and extent of WD for some time before converging on these, though.

Completely agree with your comment about the importance of parameter tuning, especially for targets that are difficult to predict! Might I ask which psychological variable(s) this applied to in your case/study? I certainly am hopeful that in time better performances might come with any harmony of more data/better optimisations/better architectures.

Best wishes,
James
